sycl/include/sycl/ext/oneapi/get_kernel_info.hpp

```diff
@@ -0,0 +1,48 @@
+//==----- get_kernel_info.hpp --- SYCL get_kernel_info extension -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------===//
+
+#pragma once
+#include <sycl/context.hpp>
+#include <sycl/detail/export.hpp>
+#include <sycl/detail/info_desc_helpers.hpp>
+#include <sycl/device.hpp>
+#include <sycl/queue.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+
+template <typename KernelName, typename Param>
+typename sycl::detail::is_kernel_info_desc<Param>::return_type
+get_kernel_info(const context &Ctx) {
+  auto Bundle =
+      sycl::get_kernel_bundle<KernelName, sycl::bundle_state::executable>(Ctx);
+  return Bundle.template get_kernel<KernelName>().template get_info<Param>();
+}
+
+template <typename KernelName, typename Param>
+typename sycl::detail::is_kernel_device_specific_info_desc<Param>::return_type
+get_kernel_info(const context &Ctx, const device &Dev) {
+  auto Bundle =
+      sycl::get_kernel_bundle<KernelName, sycl::bundle_state::executable>(Ctx);
+  return Bundle.template get_kernel<KernelName>().template get_info<Param>(Dev);
+}
+
+template <typename KernelName, typename Param>
+typename sycl::detail::is_kernel_device_specific_info_desc<Param>::return_type
+get_kernel_info(const queue &Q) {
```

Comment:
Can we extend of the test cases to also cover this overload of `get_kernel_info`? Currently, the tests only cover the other two overloads.

Reply:
Good point, will do.